UI/MemStickScreen.cpp

```diff
@@ -51,45 +111,50 @@ void MemStickScreen::CreateViews() {
 
 	Margins actionMenuMargins(15, 15, 15, 0);
 
-	root_ = new AnchorLayout();
-
-	ViewGroup *columns = new LinearLayout(ORIENT_HORIZONTAL, new AnchorLayoutParams(FILL_PARENT, FILL_PARENT));
-	root_->Add(columns);
+	root_ = new LinearLayout(ORIENT_HORIZONTAL);
 
-	ViewGroup *leftColumn = new AnchorLayout(new LinearLayoutParams(1.0));
+	Spacer *spacerColumn = new Spacer(new LinearLayoutParams(20.0, FILL_PARENT, 0.0f));
+	ViewGroup *leftColumn = new LinearLayout(ORIENT_VERTICAL, new LinearLayoutParams(1.0));
 	ViewGroup *rightColumnItems = new LinearLayout(ORIENT_VERTICAL, new LinearLayoutParams(300, FILL_PARENT, actionMenuMargins));
-	columns->Add(leftColumn);
-	columns->Add(rightColumnItems);
-
-	Path path = g_Config.memStickDirectory;
-	int64_t freeSpaceAtMemStick = -1;
-#if PPSSPP_PLATFORM(ANDROID)
-	if (Android_IsContentUri(path.ToString())) {
-		freeSpaceAtMemStick = Android_GetFreeSpaceByContentUri(path.ToString());
-	} else {
-		freeSpaceAtMemStick = Android_GetFreeSpaceByFilePath(path.ToString());
+	root_->Add(spacerColumn);
+	root_->Add(leftColumn);
+	root_->Add(rightColumnItems);
+
+	if (initialSetup_) {
+		leftColumn->Add(new TextView(iz->T("Welcome to PPSSPP!"), ALIGN_LEFT, false));
+		leftColumn->Add(new Spacer(new LinearLayoutParams(FILL_PARENT, 12.0f, 0.0f)));
 	}
-#endif
 
-	int leftSide = 100;
-	settingInfo_ = new SettingInfoMessage(ALIGN_CENTER | FLAG_WRAP_TEXT, new AnchorLayoutParams(dp_xres - leftSide - 40.0f, WRAP_CONTENT, leftSide, dp_yres - 80.0f - 40.0f, NONE, NONE));
-	settingInfo_->SetBottomCutoff(dp_yres - 200.0f);
+	leftColumn->Add(new TextView(iz->T("MemoryStickDescription", "Choose PSP data storage (Memory Stick)"), ALIGN_LEFT, false));
+
+	leftColumn->Add(new Choice(iz->T("Create or Choose a PSP folder")))->OnClick.Handle(this, &MemStickScreen::OnBrowse);
+	leftColumn->Add(new TextView(iz->T("ChooseFolderDesc", "* Data will stay even if you uninstall PPSSPP.\n* Data can be shared with PPSSPP Gold\n* Easy USB access"), ALIGN_LEFT, false));
 
-	root_->Add(settingInfo_);
+	leftColumn->Add(new Choice(iz->T("Use App Private Directory")))->OnClick.Handle(this, &MemStickScreen::OnUseInternalStorage);
+	leftColumn->Add(new TextView(iz->T("InternalStorageDesc", "* Warning! Data will be deleted if you uninstall PPSSPP!\n* Data cannot be shared with PPSSPP Gold\n* USB access through Android/data/org.ppsspp.ppsspp/files"), ALIGN_LEFT, false));
```

Comment:
How well does the ini parser handle embedded newlines?

-[Unknown]

Reply:
It does handle it, if I remember correctly. but better check....

Reply:
yeah, at least on the right side, heh.